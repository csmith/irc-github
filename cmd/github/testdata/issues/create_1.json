{
  "action": "opened",
  "issue": {
    "url": "https://api.github.com/repos/greboid/irc/issues/5",
    "repository_url": "https://api.github.com/repos/greboid/irc",
    "labels_url": "https://api.github.com/repos/greboid/irc/issues/5/labels{/name}",
    "comments_url": "https://api.github.com/repos/greboid/irc/issues/5/comments",
    "events_url": "https://api.github.com/repos/greboid/irc/issues/5/events",
    "html_url": "https://github.com/greboid/irc/issues/5",
    "id": 584325274,
    "node_id": "MDU6SXNzdWU1ODQzMjUyNzQ=",
    "number": 5,
    "title": "[BUG] GitHub: new branches say \"from <nothing>\"",
    "user": {
      "login": "csmith",
      "id": 189133,
      "node_id": "MDQ6VXNlcjE4OTEzMw==",
      "avatar_url": "https://avatars0.githubusercontent.com/u/189133?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/csmith",
      "html_url": "https://github.com/csmith",
      "followers_url": "https://api.github.com/users/csmith/followers",
      "following_url": "https://api.github.com/users/csmith/following{/other_user}",
      "gists_url": "https://api.github.com/users/csmith/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/csmith/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/csmith/subscriptions",
      "organizations_url": "https://api.github.com/users/csmith/orgs",
      "repos_url": "https://api.github.com/users/csmith/repos",
      "events_url": "https://api.github.com/users/csmith/events{/privacy}",
      "received_events_url": "https://api.github.com/users/csmith/received_events",
      "type": "User",
      "site_admin": false
    },
    "labels": [
      {
        "id": 1873189829,
        "node_id": "MDU6TGFiZWwxODczMTg5ODI5",
        "url": "https://api.github.com/repos/greboid/irc/labels/bug",
        "name": "bug",
        "color": "d73a4a",
        "default": true,
        "description": "Something isn't working"
      }
    ],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [

    ],
    "milestone": null,
    "comments": 0,
    "created_at": "2020-03-19T10:53:27Z",
    "updated_at": "2020-03-19T10:53:27Z",
    "closed_at": null,
    "author_association": "CONTRIBUTOR",
    "body": "When dependabot creates a new branch, the message on IRC looks like:\r\n\r\n> [simpicapp/simpic] dependabot-preview[bot] created branch dependabot/npm_and_yarn/frontend/mnemonist-0.34.0 from - https://github.com/simpicapp/simpic/commit/dcb6c41b4c84\r\n\r\n<details>\r\n<summary>JSON payload</summary>\r\n\r\n```json\r\n{\r\n  \"ref\": \"refs/heads/dependabot/npm_and_yarn/frontend/mnemonist-0.34.0\",\r\n  \"before\": \"0000000000000000000000000000000000000000\",\r\n  \"after\": \"dcb6c41b4c842e7535825f2fb7c029c76af4783e\",\r\n  \"repository\": {\r\n    \"id\": 240111851,\r\n    \"node_id\": \"MDEwOlJlcG9zaXRvcnkyNDAxMTE4NTE=\",\r\n    \"name\": \"simpic\",\r\n    \"full_name\": \"simpicapp/simpic\",\r\n    \"private\": false,\r\n    \"owner\": {\r\n      \"name\": \"simpicapp\",\r\n      \"email\": null,\r\n      \"login\": \"simpicapp\",\r\n      \"id\": 61029506,\r\n      \"node_id\": \"MDEyOk9yZ2FuaXphdGlvbjYxMDI5NTA2\",\r\n      \"avatar_url\": \"https://avatars2.githubusercontent.com/u/61029506?v=4\",\r\n      \"gravatar_id\": \"\",\r\n      \"url\": \"https://api.github.com/users/simpicapp\",\r\n      \"html_url\": \"https://github.com/simpicapp\",\r\n      \"followers_url\": \"https://api.github.com/users/simpicapp/followers\",\r\n      \"following_url\": \"https://api.github.com/users/simpicapp/following{/other_user}\",\r\n      \"gists_url\": \"https://api.github.com/users/simpicapp/gists{/gist_id}\",\r\n      \"starred_url\": \"https://api.github.com/users/simpicapp/starred{/owner}{/repo}\",\r\n      \"subscriptions_url\": \"https://api.github.com/users/simpicapp/subscriptions\",\r\n      \"organizations_url\": \"https://api.github.com/users/simpicapp/orgs\",\r\n      \"repos_url\": \"https://api.github.com/users/simpicapp/repos\",\r\n      \"events_url\": \"https://api.github.com/users/simpicapp/events{/privacy}\",\r\n      \"received_events_url\": \"https://api.github.com/users/simpicapp/received_events\",\r\n      \"type\": \"Organization\",\r\n      \"site_admin\": false\r\n    },\r\n    \"html_url\": \"https://github.com/simpicapp/simpic\",\r\n    \"description\": \"Simple self-hosted photo manager\",\r\n    \"fork\": false,\r\n    \"url\": \"https://github.com/simpicapp/simpic\",\r\n    \"forks_url\": \"https://api.github.com/repos/simpicapp/simpic/forks\",\r\n    \"keys_url\": \"https://api.github.com/repos/simpicapp/simpic/keys{/key_id}\",\r\n    \"collaborators_url\": \"https://api.github.com/repos/simpicapp/simpic/collaborators{/collaborator}\",\r\n    \"teams_url\": \"https://api.github.com/repos/simpicapp/simpic/teams\",\r\n    \"hooks_url\": \"https://api.github.com/repos/simpicapp/simpic/hooks\",\r\n    \"issue_events_url\": \"https://api.github.com/repos/simpicapp/simpic/issues/events{/number}\",\r\n    \"events_url\": \"https://api.github.com/repos/simpicapp/simpic/events\",\r\n    \"assignees_url\": \"https://api.github.com/repos/simpicapp/simpic/assignees{/user}\",\r\n    \"branches_url\": \"https://api.github.com/repos/simpicapp/simpic/branches{/branch}\",\r\n    \"tags_url\": \"https://api.github.com/repos/simpicapp/simpic/tags\",\r\n    \"blobs_url\": \"https://api.github.com/repos/simpicapp/simpic/git/blobs{/sha}\",\r\n    \"git_tags_url\": \"https://api.github.com/repos/simpicapp/simpic/git/tags{/sha}\",\r\n    \"git_refs_url\": \"https://api.github.com/repos/simpicapp/simpic/git/refs{/sha}\",\r\n    \"trees_url\": \"https://api.github.com/repos/simpicapp/simpic/git/trees{/sha}\",\r\n    \"statuses_url\": \"https://api.github.com/repos/simpicapp/simpic/statuses/{sha}\",\r\n    \"languages_url\": \"https://api.github.com/repos/simpicapp/simpic/languages\",\r\n    \"stargazers_url\": \"https://api.github.com/repos/simpicapp/simpic/stargazers\",\r\n    \"contributors_url\": \"https://api.github.com/repos/simpicapp/simpic/contributors\",\r\n    \"subscribers_url\": \"https://api.github.com/repos/simpicapp/simpic/subscribers\",\r\n    \"subscription_url\": \"https://api.github.com/repos/simpicapp/simpic/subscription\",\r\n    \"commits_url\": \"https://api.github.com/repos/simpicapp/simpic/commits{/sha}\",\r\n    \"git_commits_url\": \"https://api.github.com/repos/simpicapp/simpic/git/commits{/sha}\",\r\n    \"comments_url\": \"https://api.github.com/repos/simpicapp/simpic/comments{/number}\",\r\n    \"issue_comment_url\": \"https://api.github.com/repos/simpicapp/simpic/issues/comments{/number}\",\r\n    \"contents_url\": \"https://api.github.com/repos/simpicapp/simpic/contents/{+path}\",\r\n    \"compare_url\": \"https://api.github.com/repos/simpicapp/simpic/compare/{base}...{head}\",\r\n    \"merges_url\": \"https://api.github.com/repos/simpicapp/simpic/merges\",\r\n    \"archive_url\": \"https://api.github.com/repos/simpicapp/simpic/{archive_format}{/ref}\",\r\n    \"downloads_url\": \"https://api.github.com/repos/simpicapp/simpic/downloads\",\r\n    \"issues_url\": \"https://api.github.com/repos/simpicapp/simpic/issues{/number}\",\r\n    \"pulls_url\": \"https://api.github.com/repos/simpicapp/simpic/pulls{/number}\",\r\n    \"milestones_url\": \"https://api.github.com/repos/simpicapp/simpic/milestones{/number}\",\r\n    \"notifications_url\": \"https://api.github.com/repos/simpicapp/simpic/notifications{?since,all,participating}\",\r\n    \"labels_url\": \"https://api.github.com/repos/simpicapp/simpic/labels{/name}\",\r\n    \"releases_url\": \"https://api.github.com/repos/simpicapp/simpic/releases{/id}\",\r\n    \"deployments_url\": \"https://api.github.com/repos/simpicapp/simpic/deployments\",\r\n    \"created_at\": 1581540441,\r\n    \"updated_at\": \"2020-03-18T11:40:56Z\",\r\n    \"pushed_at\": 1584614397,\r\n    \"git_url\": \"git://github.com/simpicapp/simpic.git\",\r\n    \"ssh_url\": \"git@github.com:simpicapp/simpic.git\",\r\n    \"clone_url\": \"https://github.com/simpicapp/simpic.git\",\r\n    \"svn_url\": \"https://github.com/simpicapp/simpic\",\r\n    \"homepage\": \"https://simpic.app/\",\r\n    \"size\": 3002,\r\n    \"stargazers_count\": 2,\r\n    \"watchers_count\": 2,\r\n    \"language\": \"Go\",\r\n    \"has_issues\": true,\r\n    \"has_projects\": false,\r\n    \"has_downloads\": true,\r\n    \"has_wiki\": false,\r\n    \"has_pages\": false,\r\n    \"forks_count\": 1,\r\n    \"mirror_url\": null,\r\n    \"archived\": false,\r\n    \"disabled\": false,\r\n    \"open_issues_count\": 22,\r\n    \"license\": {\r\n      \"key\": \"other\",\r\n      \"name\": \"Other\",\r\n      \"spdx_id\": \"NOASSERTION\",\r\n      \"url\": null,\r\n      \"node_id\": \"MDc6TGljZW5zZTA=\"\r\n    },\r\n    \"forks\": 1,\r\n    \"open_issues\": 22,\r\n    \"watchers\": 2,\r\n    \"default_branch\": \"master\",\r\n    \"stargazers\": 2,\r\n    \"master_branch\": \"master\",\r\n    \"organization\": \"simpicapp\"\r\n  },\r\n  \"pusher\": {\r\n    \"name\": \"dependabot-preview[bot]\",\r\n    \"email\": null\r\n  },\r\n  \"organization\": {\r\n    \"login\": \"simpicapp\",\r\n    \"id\": 61029506,\r\n    \"node_id\": \"MDEyOk9yZ2FuaXphdGlvbjYxMDI5NTA2\",\r\n    \"url\": \"https://api.github.com/orgs/simpicapp\",\r\n    \"repos_url\": \"https://api.github.com/orgs/simpicapp/repos\",\r\n    \"events_url\": \"https://api.github.com/orgs/simpicapp/events\",\r\n    \"hooks_url\": \"https://api.github.com/orgs/simpicapp/hooks\",\r\n    \"issues_url\": \"https://api.github.com/orgs/simpicapp/issues\",\r\n    \"members_url\": \"https://api.github.com/orgs/simpicapp/members{/member}\",\r\n    \"public_members_url\": \"https://api.github.com/orgs/simpicapp/public_members{/member}\",\r\n    \"avatar_url\": \"https://avatars2.githubusercontent.com/u/61029506?v=4\",\r\n    \"description\": \"\"\r\n  },\r\n  \"sender\": {\r\n    \"login\": \"dependabot-preview[bot]\",\r\n    \"id\": 27856297,\r\n    \"node_id\": \"MDM6Qm90Mjc4NTYyOTc=\",\r\n    \"avatar_url\": \"https://avatars3.githubusercontent.com/in/2141?v=4\",\r\n    \"gravatar_id\": \"\",\r\n    \"url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D\",\r\n    \"html_url\": \"https://github.com/apps/dependabot-preview\",\r\n    \"followers_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/followers\",\r\n    \"following_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/following{/other_user}\",\r\n    \"gists_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/gists{/gist_id}\",\r\n    \"starred_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/starred{/owner}{/repo}\",\r\n    \"subscriptions_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/subscriptions\",\r\n    \"organizations_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/orgs\",\r\n    \"repos_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/repos\",\r\n    \"events_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/events{/privacy}\",\r\n    \"received_events_url\": \"https://api.github.com/users/dependabot-preview%5Bbot%5D/received_events\",\r\n    \"type\": \"Bot\",\r\n    \"site_admin\": false\r\n  },\r\n  \"created\": true,\r\n  \"deleted\": false,\r\n  \"forced\": false,\r\n  \"base_ref\": null,\r\n  \"compare\": \"https://github.com/simpicapp/simpic/commit/dcb6c41b4c84\",\r\n  \"commits\": [\r\n    {\r\n      \"id\": \"dcb6c41b4c842e7535825f2fb7c029c76af4783e\",\r\n      \"tree_id\": \"5aeba54d848fe0ad0c0e029f5a17193de65df66c\",\r\n      \"distinct\": true,\r\n      \"message\": \"Bump mnemonist from 0.33.1 to 0.34.0 in /frontend\\n\\nBumps [mnemonist](https://github.com/yomguithereal/mnemonist) from 0.33.1 to 0.34.0.\\n- [Release notes](https://github.com/yomguithereal/mnemonist/releases)\\n- [Changelog](https://github.com/Yomguithereal/mnemonist/blob/master/CHANGELOG.md)\\n- [Commits](https://github.com/yomguithereal/mnemonist/compare/0.33.1...0.34.0)\\n\\nSigned-off-by: dependabot-preview[bot] <support@dependabot.com>\",\r\n      \"timestamp\": \"2020-03-19T10:39:56Z\",\r\n      \"url\": \"https://github.com/simpicapp/simpic/commit/dcb6c41b4c842e7535825f2fb7c029c76af4783e\",\r\n      \"author\": {\r\n        \"name\": \"dependabot-preview[bot]\",\r\n        \"email\": \"27856297+dependabot-preview[bot]@users.noreply.github.com\",\r\n        \"username\": \"dependabot-preview[bot]\"\r\n      },\r\n      \"committer\": {\r\n        \"name\": \"GitHub\",\r\n        \"email\": \"noreply@github.com\",\r\n        \"username\": \"web-flow\"\r\n      },\r\n      \"added\": [\r\n\r\n      ],\r\n      \"removed\": [\r\n\r\n      ],\r\n      \"modified\": [\r\n        \"frontend/package-lock.json\",\r\n        \"frontend/package.json\"\r\n      ]\r\n    }\r\n  ],\r\n  \"head_commit\": {\r\n    \"id\": \"dcb6c41b4c842e7535825f2fb7c029c76af4783e\",\r\n    \"tree_id\": \"5aeba54d848fe0ad0c0e029f5a17193de65df66c\",\r\n    \"distinct\": true,\r\n    \"message\": \"Bump mnemonist from 0.33.1 to 0.34.0 in /frontend\\n\\nBumps [mnemonist](https://github.com/yomguithereal/mnemonist) from 0.33.1 to 0.34.0.\\n- [Release notes](https://github.com/yomguithereal/mnemonist/releases)\\n- [Changelog](https://github.com/Yomguithereal/mnemonist/blob/master/CHANGELOG.md)\\n- [Commits](https://github.com/yomguithereal/mnemonist/compare/0.33.1...0.34.0)\\n\\nSigned-off-by: dependabot-preview[bot] <support@dependabot.com>\",\r\n    \"timestamp\": \"2020-03-19T10:39:56Z\",\r\n    \"url\": \"https://github.com/simpicapp/simpic/commit/dcb6c41b4c842e7535825f2fb7c029c76af4783e\",\r\n    \"author\": {\r\n      \"name\": \"dependabot-preview[bot]\",\r\n      \"email\": \"27856297+dependabot-preview[bot]@users.noreply.github.com\",\r\n      \"username\": \"dependabot-preview[bot]\"\r\n    },\r\n    \"committer\": {\r\n      \"name\": \"GitHub\",\r\n      \"email\": \"noreply@github.com\",\r\n      \"username\": \"web-flow\"\r\n    },\r\n    \"added\": [\r\n\r\n    ],\r\n    \"removed\": [\r\n\r\n    ],\r\n    \"modified\": [\r\n      \"frontend/package-lock.json\",\r\n      \"frontend/package.json\"\r\n    ]\r\n  }\r\n}\r\n```\r\n</details>"
  },
  "repository": {
    "id": 243344149,
    "node_id": "MDEwOlJlcG9zaXRvcnkyNDMzNDQxNDk=",
    "name": "irc",
    "full_name": "greboid/irc",
    "private": false,
    "owner": {
      "login": "greboid",
      "id": 189618,
      "node_id": "MDQ6VXNlcjE4OTYxOA==",
      "avatar_url": "https://avatars0.githubusercontent.com/u/189618?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/greboid",
      "html_url": "https://github.com/greboid",
      "followers_url": "https://api.github.com/users/greboid/followers",
      "following_url": "https://api.github.com/users/greboid/following{/other_user}",
      "gists_url": "https://api.github.com/users/greboid/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/greboid/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/greboid/subscriptions",
      "organizations_url": "https://api.github.com/users/greboid/orgs",
      "repos_url": "https://api.github.com/users/greboid/repos",
      "events_url": "https://api.github.com/users/greboid/events{/privacy}",
      "received_events_url": "https://api.github.com/users/greboid/received_events",
      "type": "User",
      "site_admin": false
    },
    "html_url": "https://github.com/greboid/irc",
    "description": "Plugin based IRC library for Go",
    "fork": false,
    "url": "https://api.github.com/repos/greboid/irc",
    "forks_url": "https://api.github.com/repos/greboid/irc/forks",
    "keys_url": "https://api.github.com/repos/greboid/irc/keys{/key_id}",
    "collaborators_url": "https://api.github.com/repos/greboid/irc/collaborators{/collaborator}",
    "teams_url": "https://api.github.com/repos/greboid/irc/teams",
    "hooks_url": "https://api.github.com/repos/greboid/irc/hooks",
    "issue_events_url": "https://api.github.com/repos/greboid/irc/issues/events{/number}",
    "events_url": "https://api.github.com/repos/greboid/irc/events",
    "assignees_url": "https://api.github.com/repos/greboid/irc/assignees{/user}",
    "branches_url": "https://api.github.com/repos/greboid/irc/branches{/branch}",
    "tags_url": "https://api.github.com/repos/greboid/irc/tags",
    "blobs_url": "https://api.github.com/repos/greboid/irc/git/blobs{/sha}",
    "git_tags_url": "https://api.github.com/repos/greboid/irc/git/tags{/sha}",
    "git_refs_url": "https://api.github.com/repos/greboid/irc/git/refs{/sha}",
    "trees_url": "https://api.github.com/repos/greboid/irc/git/trees{/sha}",
    "statuses_url": "https://api.github.com/repos/greboid/irc/statuses/{sha}",
    "languages_url": "https://api.github.com/repos/greboid/irc/languages",
    "stargazers_url": "https://api.github.com/repos/greboid/irc/stargazers",
    "contributors_url": "https://api.github.com/repos/greboid/irc/contributors",
    "subscribers_url": "https://api.github.com/repos/greboid/irc/subscribers",
    "subscription_url": "https://api.github.com/repos/greboid/irc/subscription",
    "commits_url": "https://api.github.com/repos/greboid/irc/commits{/sha}",
    "git_commits_url": "https://api.github.com/repos/greboid/irc/git/commits{/sha}",
    "comments_url": "https://api.github.com/repos/greboid/irc/comments{/number}",
    "issue_comment_url": "https://api.github.com/repos/greboid/irc/issues/comments{/number}",
    "contents_url": "https://api.github.com/repos/greboid/irc/contents/{+path}",
    "compare_url": "https://api.github.com/repos/greboid/irc/compare/{base}...{head}",
    "merges_url": "https://api.github.com/repos/greboid/irc/merges",
    "archive_url": "https://api.github.com/repos/greboid/irc/{archive_format}{/ref}",
    "downloads_url": "https://api.github.com/repos/greboid/irc/downloads",
    "issues_url": "https://api.github.com/repos/greboid/irc/issues{/number}",
    "pulls_url": "https://api.github.com/repos/greboid/irc/pulls{/number}",
    "milestones_url": "https://api.github.com/repos/greboid/irc/milestones{/number}",
    "notifications_url": "https://api.github.com/repos/greboid/irc/notifications{?since,all,participating}",
    "labels_url": "https://api.github.com/repos/greboid/irc/labels{/name}",
    "releases_url": "https://api.github.com/repos/greboid/irc/releases{/id}",
    "deployments_url": "https://api.github.com/repos/greboid/irc/deployments",
    "created_at": "2020-02-26T19:09:37Z",
    "updated_at": "2020-03-19T00:36:43Z",
    "pushed_at": "2020-03-19T00:36:41Z",
    "git_url": "git://github.com/greboid/irc.git",
    "ssh_url": "git@github.com:greboid/irc.git",
    "clone_url": "https://github.com/greboid/irc.git",
    "svn_url": "https://github.com/greboid/irc",
    "homepage": "",
    "size": 168,
    "stargazers_count": 0,
    "watchers_count": 0,
    "language": "Go",
    "has_issues": true,
    "has_projects": true,
    "has_downloads": true,
    "has_wiki": true,
    "has_pages": false,
    "forks_count": 1,
    "mirror_url": null,
    "archived": false,
    "disabled": false,
    "open_issues_count": 1,
    "license": {
      "key": "mit",
      "name": "MIT License",
      "spdx_id": "MIT",
      "url": "https://api.github.com/licenses/mit",
      "node_id": "MDc6TGljZW5zZTEz"
    },
    "forks": 1,
    "open_issues": 1,
    "watchers": 0,
    "default_branch": "master"
  },
  "sender": {
    "login": "csmith",
    "id": 189133,
    "node_id": "MDQ6VXNlcjE4OTEzMw==",
    "avatar_url": "https://avatars0.githubusercontent.com/u/189133?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/csmith",
    "html_url": "https://github.com/csmith",
    "followers_url": "https://api.github.com/users/csmith/followers",
    "following_url": "https://api.github.com/users/csmith/following{/other_user}",
    "gists_url": "https://api.github.com/users/csmith/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/csmith/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/csmith/subscriptions",
    "organizations_url": "https://api.github.com/users/csmith/orgs",
    "repos_url": "https://api.github.com/users/csmith/repos",
    "events_url": "https://api.github.com/users/csmith/events{/privacy}",
    "received_events_url": "https://api.github.com/users/csmith/received_events",
    "type": "User",
    "site_admin": false
  }
}