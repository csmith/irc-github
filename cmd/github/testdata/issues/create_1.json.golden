[greboid/irc] csmith create issue: [BUG] GitHub: new branches say "from <nothing>" -  https://github.com/greboid/irc/issues/5